functions/ask.js

```
@@ -0,0 +1,155 @@
+                const data = treeData[receivedMessage]
+                return {
+                    statusCode: 200,
+                    body: JSON.stringify({
+                        'result': data,
+                        'statusCode': 200
+                    }
+                    )
+                }
+            }
+            else {
+                const allTreeDataKeys = Object.keys(treeData)
+                const allOptionalUtterances = []
+
+                const dock = await dockStart({ use: ['Basic'], autoSave: false, modelFileName: '/tmp/model.nlp' });
+                const nlp = dock.get('nlp');
+
+                nlp.load('/tmp/model.nlp')
+                nlp.addLanguage('nl');
+
+                allTreeDataKeys.forEach(key => {
+                    if (treeData[key].optionalUtterances && treeData[key].optionalUtterances.length) {
+                        treeData[key].optionalUtterances.forEach(utterance => {
+                            allOptionalUtterances.push(
+                                {
+                                    message: treeData[key].message,
+                                    intent: key,
+                                    utterance: utterance
+                                }
+                            )  
+                        })
+                    }
+                })
+                
+                allOptionalUtterances.forEach(item => {
+                    const language = 'NL'
+                    const currentIntent = item.intent
+                    const currentUtterance = item.utterance
+                    const currentAnswer = item.message
+
+                    nlp.addDocument(language, currentUtterance, currentIntent);
+                    nlp.addAnswer(language, currentIntent, currentAnswer);
+                })
+
+                try {
+                    await nlp.train()
+                }
+                catch (error) {
+                    console.log(error);
```

You are swallowing an error here. Are you sure about this?